File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<int>> same_word_count;
        int n=strs.size();
        for(int i=0; i<n; i++){
            string sorted_word = strs[i];
            sort(sorted_word.begin(), sorted_word.end());
            same_word_count[sorted_word].push_back(i);
        }
        vector<vector<string>> ans;
        for(auto it: same_word_count){
            vector<string> tmp;
            for(auto ind: it.second){
                tmp.push_back(strs[ind]);
            }
            ans.push_back(tmp);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anargam_map = defaultdict(list)
        for i in range(len(strs)):
            char_count = [0]*26
            for j in strs[i]:
                char_count[ord(j)-ord('a')]+=1
            if tuple(char_count) in anargam_map:
                anargam_map[tuple(char_count)].append(strs[i])
            else:
                anargam_map[tuple(char_count)]=[strs[i]]
        
        return list(anargam_map.values())